seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -0,0 +1,177 @@
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()['error'], 'Failed to grant seqr service account access to the workspace {}/{}'
+                              .format(TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME))
+
+        # Test valid operation
+        mock_add_service_account.reset_mock(side_effect=True)
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        self.assertEqual(response.status_code, 200)
+        project = Project.objects.get(workspace_namespace=TEST_WORKSPACE_NAMESPACE, workspace_name=TEST_NO_PROJECT_WORKSPACE_NAME)
+        response_json = response.json()
+        self.assertEqual(project.guid, response_json['projectGuid'])
+        self.assertListEqual(
+            [project.genome_version, project.description, project.workspace_namespace, project.workspace_name],
+            ['38', 'A test project', TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME])
+        mock_add_service_account.assert_called_with(user, TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME)
+        mock_email.assert_called_with(
+            subject='AnVIL data loading request',
+            body="""
+    Data from AnVIL workspace "{namespace}/{name}" needs to be loaded to seqr project <a href="http://testserver/project/{guid}/project_page">{name}</a> (guid: {guid})
+
+    The sample IDs to load are attached.    
+    """.format(namespace=TEST_WORKSPACE_NAMESPACE, name=TEST_NO_PROJECT_WORKSPACE_NAME, guid=project.guid),
+            to=['test_superuser@test.com', 'test_data_manager@test.com'],
+            attachments=[('{}_sample_ids.tsv'.format(project.guid), 'NA19675\nNA19678\nHG00735')]
+        )
+
+        # Test project exist
+        url = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME])
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Project "{name}" for workspace "{namespace}/{name}" exists.'
+                         .format(namespace=TEST_WORKSPACE_NAMESPACE, name=TEST_NO_PROJECT_WORKSPACE_NAME))
+
+        # Test sending email exception
+        url = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME2])
+        mock_email.side_effect = Exception('Something wrong while sending email.')
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        self.assertEqual(response.status_code, 200)
+        mock_api_logger.error.assert_called_with('Exception while sending email to user test_user_manager. Something wrong while sending email.')
+
+        # Test login locally
+        remove_token(user)  # The user will be same as logging in locally after the access token is removed
```

Comment:
this should tested by the local test case below and isn't necessary

Reply:
The code coverage is slightly different from that of the test in the local-only auth mode. But it is true that it is not necessary if `is_google_authenticated()` has already been tested.